### PR TITLE
FE: Monitoring Policies - robust unit tests for monitoringPoliciesStore

### PR DESCRIPTION
**Pull request by faizanjanjua**
## Description
Update the unit tests for monitoringPoliciesStore to test actual functionality, for example that deleteRule deletes the rule with the given id from the selectedPolicy, etc.

An example is here: [Example of monitoringPoliciesStore unit test by synqotik · Pull Request #1849 · OpenNMS-Cloud/lokahi](https://github.com/OpenNMS-Cloud/lokahi/pull/1849)

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2504

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/tests/store/Views/monitoringPoliciesStore.test.ts [hunk 1]
```diff
@@ -237,46 +250,318 @@ describe('Monitoring Policies Store', () => {
...
+          alertConditions: [],
+          id: 1,
+          name: 'Dummy Rule',
+          thresholdMetricName: 'Dummy Metric'
+        }]})
+        store.selectedPolicy = mockMonitoringPolicy
+        store.selectedRule = (dummy &&  dummy.rules !== undefined) ? {...dummy.rules[0]} : {}
+        await store.addNewCondition()
+
+        if (store.selectedRule.alertConditions && store.selectedRule.alertConditions?.length > 0) {
```
Comment (synqotik) [inline ui/tests/store/Views/monitoringPoliciesStore.test.ts:428]: We can check that `alertConditions` now have 1 item:

```
expect(store.selectedRule.alertConditions.length).toBe(1)
```

File: ui/tests/store/Views/monitoringPoliciesStore.test.ts [hunk 2]
```diff
@@ -237,46 +250,318 @@ describe('Monitoring Policies Store', () => {
...
```
Comment (synqotik) [inline ui/tests/store/Views/monitoringPoliciesStore.test.ts:449]: See above, can check there is 1 condition now

File: ui/tests/store/Views/monitoringPoliciesStore.test.ts [hunk 3]
```diff
@@ -237,46 +250,318 @@ describe('Monitoring Policies Store', () => {
...
+
       store.selectedPolicy = {...mockMonitoringPolicy, isDefault: false}
-      store.validateMonitoringPolicy(store.selectedPolicy)
-      expect(store.validateMonitoringPolicy).toHaveBeenCalledWith(store.selectedPolicy)
-      expect(store.validateMonitoringPolicy).toBeTruthy()
+      store.selectedRule = mockMonitoringPolicy && mockMonitoringPolicy.rules ? mockMonitoringPolicy?.rules[0] : {}
+
+      await store.removeRule()
+      expect(store.selectedRule).toBeUndefined()
+      expect(store.selectedPolicy.rules).length.lessThanOrEqual(0)
```
Comment (synqotik) [inline ui/tests/store/Views/monitoringPoliciesStore.test.ts:464]: `length` should now be 0, can do:

```
expect(store.selectedPolicy.rules.length).toBe(0)
```

**Review by synqotik — CHANGES_REQUESTED** [review]
Overall looks good. Added a couple cases where we can check a few more things.

File: ui/tests/store/Views/monitoringPoliciesStore.test.ts (hunk 3 above)
Comment (faizanjanjua) [reply ui/tests/store/Views/monitoringPoliciesStore.test.ts:464]: `  await store.removeRule()
      expect(store.selectedRule).toBeUndefined()
       // @ts-ignore
      expect(store.selectedPolicy.rules.length).toBe(0)`
      
      
      i have changed it to this.. but for this we have to use @ts-ignore Warning 
      
      other possible solution could be 
     ` await store.removeRule()
      expect(store.selectedRule).toBeUndefined()
      if(store.selectedPolicy.rules !==undefined) {
      expect(store.selectedPolicy.rules.length).toBe(0)
      }`
      
      suggest me better option so  i can change and commit.

**Review by synqotik — APPROVED** [review]
LGTM!

**Pull request merged**